File: C++/source/Framework/Messaging/LightHouse/Test/LightHouseTestLightPulse1.cpp
/***************************************************************************
 *   Copyright (C) 2005 Vidyasagara Guntaka                                *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/Test/LightHouseTestLightPulse1.h"

namespace WaveNs
{

LightHouseTestLightPulse1::LightHouseTestLightPulse1 ()
    : LightPulse (getLightPulseName ())
{
}

LightHouseTestLightPulse1::~LightHouseTestLightPulse1 ()
{
}

string LightHouseTestLightPulse1::getLightPulseName ()
{
    return ("LightHouseTestLightPulse1");
}

string LightHouseTestLightPulse1::getLightMessage () const
{
    return (m_lightMessage);
}

void LightHouseTestLightPulse1::setLightMessage (const string &lightMessage)
{
    m_lightMessage = lightMessage;
}

void LightHouseTestLightPulse1::setupAttributesForSerialization ()
{
    LightPulse::setupAttributesForSerialization ();

    addSerializableAttribute (new AttributeString (&m_lightMessage, "lightMessage"));
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/Test/LightHouseTestLightPulse1.java
/*************************************************************************************************************************
 * Copyright (C) 2015-2016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse.Test;

import com.CxWave.Wave.Framework.Messaging.LightHouse.LightPulse;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.OwnerOM;

@OwnerOM (om = LightHouseTestObjectManager2.class)
public class LightHouseTestLightPulse1 extends LightPulse
{
    private String m_lightMessage;

    public LightHouseTestLightPulse1 ()
    {
        super (getLightPulseName ());
    }

    public static String getLightPulseName ()
    {
        return ("LightHouseTestLightPulse1");
    }

    public String getLightMessage ()
    {
        return m_lightMessage;
    }

    public void setLightMessage (final String lightMessage)
    {
        m_lightMessage = lightMessage;
    }
}
